### DISCUSSION TITLE
cgrates integration

### DISCUSSION THREAD

**Discussion by mwolff44** [mwolff44/pyfreebilling#186] [Ideas]
We discussed with @arsenieciprian about cgrates integration with the new version of PyFreeBilling.
The first step will be be to define the features we want to integrate as the beginning. Only the mandatory ones to start playing with it.
For the technically point, I prefer to use directly their APIs.

**Comment by arsenieciprian** [comment]
i will put heare one minimal list of features that i have tested with cgrates and are working .
1. sip trunk with authorization by ip /username and password . This one i have tested and woks from atributes section togheder with kamailio directley
2. rating calls by destination tarrif plans . The logic is described by nick heare . https://nickvsnetworking.com/cgrates-in-baby-steps-part-2/
3. assign did to trunks and bill calls made by users.
I want to participate on testing . I can put on dispozition servers for testing purposes DID's ,outgoing routes, etc.

**Comment by arsenieciprian** [comment]
![IMG_0342](https://github.com/mwolff44/pyfreebilling/assets/113825451/907a0df3-eb37-4a5d-990e-23fd02b87e24)

**Reply by mwolff44** [reply]
do you know the corresponding APIs ?

**Reply by arsenieciprian** [reply]
I have configured only by csv on cgrates but we can ask on community google groups, the cgrates community is verry active and we recove verry fast answers.

**Comment by arsenieciprian** [comment]
https://github.com/cgrates/cgrates/blob/master/data/postman/Voiceland/CGRates.postman_collection_v2.1.json#L9

**Reply by mwolff44** [reply]
super thanks

**Comment by arsenieciprian** [comment]
welcome

**Reply by arsenieciprian** [reply]
If you want me to do anything for you just tell me here.

**Comment by arsenieciprian** [comment]
Hi theare , did u have decided to use cgrates as billing engine for pyfreebilling ?

**Reply by mwolff44** [reply]
I think it is a good idea, but I need some time because as cgrates is complex

**Comment by arsenieciprian** [comment]
Hi there. This platform already use cgrates.

https://github.com/irontec/ivozprovider/

**Reply by mwolff44** [reply]
Yes, a very good project developed by a good team ! but they use sql queries (if my memory is OK) and i want to only use API.
I will start with post billing, as it is the simplest. I just want to finish the current action items already planified before.

**Reply by arsenieciprian** [reply]
Perfect , let me know when i can install and start testing . Ty

**Comment by arsenieciprian** [comment]
Hi theare.
Any news about this project :) i'm ready to adopt
